src/foam/dao/index/TreeIndex.java

```diff
@@ -74,6 +86,35 @@ public FindPlan planFind(Object state, Object key) {
     return new TreeLookupFindPlan(prop_, (state != null ? ((TreeNode) state).size : 0) );
   }
 
+  public SelectPlan SelectPlanHelper(Object state, Sink sink, long skip, long limit, Comparator order, Predicate predicate) {
```

Comment:
Helper is a generic name. If you find a more meaningful name, that would be better. A convention is also to just call it select_() if it's an internal helper for select(). Also, method names typically start with lower-case characters.